common/faults/faults.h

```diff
@@ -0,0 +1,237 @@
+#define TEST_FAULT_2_LATCH_TIME 10
+#define TEST_FAULT_3_LATCH_TIME 10
+#define TEST_FAULT_4_LATCH_TIME 10
+//END AUTO LATCH DEFS
+
+//BEGIN AUTO UNLATCH DEFS
+#define BATT_FLOW_UNLATCH_TIME 10
+#define DRIVE_FLOW_UNLATCH_TIME 10
+#define MAIN_COMM_UNLATCH_TIME 10
+#define LV_DEAD_UNLATCH_TIME 10
+#define MOT_FRONT_OT_UNLATCH_TIME 10
+#define WLSPD_L_UNLATCH_TIME 10
+#define WLSPD_R_UNLATCH_TIME 10
+#define DRIVELINE_COMM_UNLATCH_TIME 10
+#define BMS_UNLATCH_TIME 10
+#define BATT_OT_UNLATCH_TIME 10
+#define TV_OFFLINE_UNLATCH_TIME 10
+#define TEST_FAULT_1_UNLATCH_TIME 10
+#define TEST_FAULT_2_UNLATCH_TIME 10
+#define TEST_FAULT_3_UNLATCH_TIME 10
+#define TEST_FAULT_4_UNLATCH_TIME 10
+//END AUTO UNLATCH DEFS
+
+//BEGIN AUTO SCREENMSG DEFS
+#define BATT_FLOW_MSG "battery pump flow brokey\0" 
+#define DRIVE_FLOW_MSG "battery pump flow brokey\0" 
+#define MAIN_COMM_MSG "main not talking\0" 
+#define LV_DEAD_MSG "LV dead, shut car off ASAP\0" 
+#define MOT_FRONT_OT_MSG "Front motors overheating\0" 
+#define WLSPD_L_MSG "wheelspeeed L broken\0" 
+#define WLSPD_R_MSG "wheelspeeed R broken\0" 
+#define DRIVELINE_COMM_MSG "Driveline offline\0" 
+#define BMS_MSG "BMS fault\0" 
+#define BATT_OT_MSG "Battery overheating\0" 
+#define TV_OFFLINE_MSG "TV offline\0" 
+#define TEST_FAULT_1_MSG "Test fault 1\0" 
+#define TEST_FAULT_2_MSG "Test fault 2\0" 
+#define TEST_FAULT_3_MSG "Test fault 3\0" 
+#define TEST_FAULT_4_MSG "Test fault 4\0" 
+//END AUTO SCREENMSG DEFS
+
+//asdf
+
+
+//BEGIN AUTO ENUM DEFS
+typedef enum {
+	MAIN_MODULE = 0,
+	DRIVELINE_FRONT = 1,
+	DASHBOARD = 2,
+	PRECHARGE = 3,
+	TV = 4,
+	TEST = 5,
+} fault_owner_t;
+//END AUTO ENUM DEFS
+
+typedef enum {
+    INFO = 0,
+    WARNING = 1,
+    CRITICAL = 2
+} fault_priority_t;
+
+
+
+//Contains info about fault state
+typedef struct {
+    bool latched;
+    int f_ID;
+} fault_message_t;
+
+//Contains info about the fault as a whole
+typedef struct {
+    bool tempLatch;
+    bool forceActive;
+    fault_priority_t priority;
+    uint8_t bounces;
+    uint16_t time_since_latch;
+    uint16_t last_rx_time;
+    int f_ID;
+    int f_max;
+    int f_min;
+    fault_message_t *message;
+    char* screen_MSG;
+} fault_attributes_t;
+
+
+
+
+//Vars
+extern fault_message_t message[TOTAL_NUM_FAULTS];
+extern fault_attributes_t attributes[TOTAL_NUM_FAULTS];
+
+//Function defs
+extern void initFaultLibrary(fault_owner_t, q_handle_t*, q_handle_t*);
+extern bool setFault(int, int);
+extern void forceFault(int, bool);
+extern void unForce(int);
+extern void txFaultSpecific(int);
+extern void txFaults();
+extern void rxFaults();
+extern void updateFaults();
+extern void recieveFaultsPeriodic();
+extern void killFaultLibrary();
+extern bool currMCULatched();
+extern bool infoLatched();
+extern bool warningLatched();
+extern bool criticalLatched();
+extern bool otherMCUsLatched();
+extern bool isLatched();
+extern bool checkFault();
+extern fault_attributes_t getFault(int id);
```

Comment:
Maybe just return whether or not the fault is latched

Reply:
I can, I thought the extra fidelity would be helpful. I was using the otherMCUsLatched() function for the nucleo when I was demoing it to you.